hadoop-client-modules/hadoop-client-runtime/pom.xml

```diff
@@ -158,11 +158,8 @@
                       <!-- the jdk ships part of the javax.annotation namespace, so if we want to relocate this we'll have to care it out by class :( -->
                       <exclude>com.google.code.findbugs:jsr305</exclude>
                       <exclude>io.dropwizard.metrics:metrics-core</exclude>
-                      <exclude>org.eclipse.jetty.websocket:*</exclude>
                       <exclude>org.eclipse.jetty:jetty-servlet</exclude>
                       <exclude>org.eclipse.jetty:jetty-security</exclude>
-                      <exclude>org.eclipse.jetty:jetty-client</exclude>
-                      <exclude>org.eclipse.jetty:jetty-http</exclude>
                       <exclude>org.eclipse.jetty:jetty-xml</exclude>
```

Comment:
Agreed that jetty-client and jetty-http are shaded as well because jetty websocket-client depends on them.

websocket-client depends on jetty-xml, too.  Should we shade jetty-xml or not?
https://github.com/eclipse/jetty.project/blob/jetty-9.4.x/jetty-websocket/websocket-client/pom.xml#L25

Reply:
jetty-servlet and jetty-security are brought by [YARN-8778](https://issues.apache.org/jira/browse/YARN-8778). jetty-client, jetty-http and jetty-xml are brought by [HADOOP-16152](https://issues.apache.org/jira/browse/HADOOP-16152). [The patch of HADOOP-16152](https://issues.apache.org/jira/secure/attachment/12982420/HADOOP-16152.006.patch#file-7) looks excluding them on purpose.

Reply:
I believe jetty-servlet and jetty-security are brought in by YARN-8763 not YARN-8778. ` `` 

> websocket-client depends on jetty-xml, too. Should we shade jetty-xml or not?

I think its fair to include jetty-xml as well in the shaded runtime jar. Will update the PR.